junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/HierarchicalTestExecutor.java

```diff
@@ -72,6 +67,25 @@ private void execute(TestDescriptor testDescriptor, C parentContext, ExecutionTr
 			return;
 		}
 
+		SkipResult skipResult;
+		try {
+			skipResult = node.shouldBeSkipped(preparedContext);
+		}
+		catch (Exception exception) {
+			rethrowIfBlacklisted(exception);
+			node.cleanup(preparedContext);
+			// We call executionStarted first to comply with the contract of EngineExecutionListener
+			this.listener.executionStarted(testDescriptor);
+			this.listener.executionFinished(testDescriptor, TestExecutionResult.failed(exception));
+			return;
+		}
+
+		if (skipResult.isSkipped()) {
+			node.cleanup(preparedContext);
```

Comment:
I think we need to report the test as failed when `cleanup()` throws an exception.

Reply:
True.

Reply:
I extracted the "reportAsFailed" logic into an own method and did some more exception handling ... cleanup.